src/Base/Parameter.cpp

```diff
@@ -1252,6 +1252,55 @@ void ParameterGrp::RemoveUnsigned(const char* Name)
     Notify(Name);
 }
 
+Base::Color ParameterGrp::GetColor(const char* Name, Base::Color lPreset) const
+{
+    auto packed = GetUnsigned(Name, lPreset.getPackedValue());
+
+    return Color(static_cast<uint32_t>(packed));
+}
+
+void ParameterGrp::SetColor(const char* Name, Base::Color lValue)
+{
+    SetUnsigned(Name, lValue.getPackedValue());
+}
+
+std::vector<Base::Color> ParameterGrp::GetColors(const char* sFilter) const
+{
+    auto packed = GetUnsigneds(sFilter);
+    std::vector<Base::Color> result;
+
+    std::transform(packed.begin(),
+                   packed.end(),
+                   std::back_inserter(result),
+                   [](const unsigned long lValue) {
+                       return Color(static_cast<uint32_t>(lValue));
+                   });
+
+    return result;
+}
+
+std::vector<std::pair<std::string, Base::Color>>
+ParameterGrp::GetColorMap(const char* sFilter) const
+{
+    auto packed = GetUnsignedMap(sFilter);
+    std::vector<std::pair<std::string, Base::Color>> result;
+
+    std::transform(packed.begin(),
+                   packed.end(),
+                   std::back_inserter(result),
+                   [](const std::pair<std::string, unsigned long>& lValue) {
+                       return std::make_pair(lValue.first,
+                                             Color(static_cast<uint32_t>(lValue.second)));
+                   });
+
+    return result;
+}

```

Comment:
This goes... though would have been better to not reiterate the result of GetUnsignedMap and double copying.

```suggestion
ParameterGrp::GetColorMap(const char* sFilter) const
{
    std::vector<std::pair<std::string, unsigned long>> vrValues;
    if (!_pGroupNode) {
        return vrValues;
    }

    std::string Name;
    const int base = 10;

    DOMElement* pcTemp = FindElement(_pGroupNode, "FCUInt");
    while (pcTemp) {
        Name = StrX(pcTemp->getAttribute(XStrLiteral("Name").unicodeForm())).c_str();
        // check on filter condition
        if (!sFilter || Name.find(sFilter) != std::string::npos) {
            uint32_t packedColor = strtoul(
                StrX(pcTemp->getAttribute(XStrLiteral("Value").unicodeForm())).c_str(),
                nullptr,
                base
            );
            vrValues.emplace_back(
                Name,
                Color(packedColor);
            );
        }
        pcTemp = FindNextElement(pcTemp, "FCUInt");
    }

    return vrValues;
}
```